packages/create-cloudflare/templates/qwik/c3.ts

```diff
@@ -35,43 +36,67 @@ const addBindingsProxy = (ctx: C3Context) => {
 	const s = spinner();
 	s.start("Updating `vite.config.ts`");
 
-	// Insert the env declaration after the last import (but before the rest of the body)
-	const envDeclaration = `
-let env = {};
-
-if(process.env.NODE_ENV === 'development') {
-  const { getPlatformProxy } = await import('wrangler');
-  const platformProxy = await getPlatformProxy();
-  env = platformProxy.env;
-}
-`;
+	const snippets = loadTemplateSnippets(ctx);
 
 	transformFile("vite.config.ts", {
+		// Insert the env declaration after the last import (but before the rest of the body)
 		visitProgram: function (n) {
 			const lastImportIndex = n.node.body.findLastIndex(
 				(t) => t.type === "ImportDeclaration"
 			);
-			n.get("body").insertAt(lastImportIndex + 1, envDeclaration);
+			const lastImport = n.get("body", lastImportIndex);
+			lastImport.insertAfter(...snippets.getPlatformProxyTs);
+
+			return this.traverse(n);
+		},
+		// Pass the `platform` object from the declaration to the `qwikCity` plugin
+		visitCallExpression: function (n) {
+			const callee = n.node.callee as recast.types.namedTypes.Identifier;
+			if (callee.name !== "qwikCity") {
+				return this.traverse(n);
+			}
+
+			n.node.arguments = [parseTs("{ platform }")];
```

Comment:
Do we need to be concerned that this plugin can take other properties on its options parameter? See https://github.com/BuilderIO/qwik/blob/main/packages/qwik-city/buildtime/vite/types.ts#L23

Here we are effectively blowing away any other properties.